autobahn/autobahn/wamp/protocol.py

```diff
@@ -481,15 +478,20 @@ def error(err):
                     raise ProtocolError("PUBLISHED received for non-pending request ID {0}".format(msg.request))
 
             elif isinstance(msg, message.Subscribed):
-
                 if msg.request in self._subscribe_reqs:
                     d, obj, fn, topic, options = self._subscribe_reqs.pop(msg.request)
-                    if options:
-                        self._subscriptions[msg.subscription] = Handler(obj, fn, topic, options.details_arg)
-                    else:
-                        self._subscriptions[msg.subscription] = Handler(obj, fn, topic)
-                    s = Subscription(self, msg.subscription)
-                    self._resolve_future(d, s)
+                    sub_id = msg.subscription
+                    # we should NEVER have our ID subscribed already
```

Comment:
Oh, okay. I'll go back to the other code I had then -- which treats self._subscriptions as as list of Handlers. So then in the tests, subscription0.id should == subscription1.id if I used the same topic, right?